apollo-client/src/main/java/com/ctrip/framework/apollo/spring/property/ScopeBeanPostProcessor.java

```diff
@@ -0,0 +1,129 @@
+import org.springframework.core.env.PropertySourcesPropertyResolver;
+
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Collection;
+
+public class ScopeBeanPostProcessor extends InstantiationAwareBeanPostProcessorAdapter implements EnvironmentAware, BeanFactoryAware {
+    private ConfigurableEnvironment environment;
+    private BeanFactory beanFactory;
+    private PropertySourcesPropertyResolver propertySourcesPropertyResolver;
+    private final PlaceholderHelper placeholderHelper;
+    private TypeConverter typeConverter;
+    private static final Logger logger = LoggerFactory.getLogger(ScopeBeanPostProcessor.class);
+
+
+    public ScopeBeanPostProcessor() {
+        this.placeholderHelper = SpringInjector.getInstance(PlaceholderHelper.class);
+        ConfigurableBeanFactory configurableBeanFactory = (ConfigurableBeanFactory) this.beanFactory;
+
+    }
+
+
+    @Override
+    public PropertyValues postProcessPropertyValues(PropertyValues pvs, PropertyDescriptor[] pds, Object bean, String beanName) throws BeansException {
+        Multimap<String, SpringValueDefinition> beanName2SpringValueDefinitions = null;
+        if (beanFactory instanceof BeanDefinitionRegistry) {
+            beanName2SpringValueDefinitions = SpringValueDefinitionProcessor
+                    .getBeanName2SpringValueDefinitions((BeanDefinitionRegistry) beanFactory);
+        }
+
+        if (propertySourcesPropertyResolver == null) {
+            propertySourcesPropertyResolver = new PropertySourcesPropertyResolver(environment.getPropertySources());
+        }
+
+        if (typeConverter == null) {
+            ConfigurableBeanFactory configurableBeanFactory = (ConfigurableBeanFactory) beanFactory;
+            this.typeConverter = configurableBeanFactory.getTypeConverter();
+
+        }
+
+
+        Collection<SpringValueDefinition> propertySpringValues = beanName2SpringValueDefinitions
+                .get(beanName);
+        if (propertySpringValues == null || propertySpringValues.isEmpty()) {
+            //不进行处理
+            return super.postProcessPropertyValues(pvs, pds, bean, beanName);
+        }
+
+        for (PropertyValue propertyValue : pvs.getPropertyValues()) {
+            //获取 placeholder
+            String placeholder = null;
+            for (SpringValueDefinition springValueDefinition : propertySpringValues) {
+                if (springValueDefinition.getPropertyName().equals(propertyValue.getName())) {
+                    placeholder = springValueDefinition.getPlaceholder();
+                    break;
+                }
+            }
+            PropertyDescriptor pd = BeanUtils
+                    .getPropertyDescriptor(bean.getClass(), propertyValue.getName());
+            Method method = pd.getWriteMethod();
+            if (method == null) {
+                continue;
+            }
+            SpringValue springValue = new SpringValue(null, placeholder,
```

Comment:
since we will handle all the beans' property values here, I think we could move the related logic of `springValueRegistry.register(beanFactory, key, springValue);` here and then we could remove the whole logic of `com.ctrip.framework.apollo.spring.annotation.SpringValueProcessor#processBeanPropertyValues`